mysql-test/suite/innodb/t/mdev-33935.test

```diff
@@ -0,0 +1,56 @@
+--source include/have_innodb.inc
+--source include/have_debug_sync.inc
```

Comment:
I would add also `include/have_debug.inc` so that the test can be skipped more quickly on a non-debug build.

Even better would be to write this test without `DEBUG_SYNC` instrumentation, so that it could run on all target architectures. We only have a few debug instrumented builds, not covering all platforms.

There already exists a test `innodb.deadlock_detect`. Could you extend it so that it demonstrates the problem?

Reply:
Based on your suggestion, there is no need to strictly align the test name with mdev-$bug -> mdev-$bug.test, so I decided to adapt the existing test ```deadlock_detect```. For me this is not so graceful, because it checking "parallel" functionality, but in the same logical scope. And if it's okey - it's okey.